ern/graphql.go

```diff
@@ -0,0 +1,553 @@
+		role: String!
+	}
+
+	type SoundRecording {
+		artistName: String!
+		# contributors: [ResourceContributor]!
+		genre: String!
+		# parentalWarningType: String
+		resourceReference: String!
+		# subGenre: String
+		soundRecordingId: String!
+		territoryCode: String!
+		title: String!
+	}
+
+	type Release {
+		artistName: String!
+		displayTitle: String!
+		releaseId: String
+		genre: String
+		releaseType: String
+	}
+`
+
+// partyDetailsArgs query arguments for PartyDetails query
+type partyDetailsArgs struct {
+	Name *string
+	ID   *string
+}
+
+// DecodeResolverObj finds a value within a MetaObject and decode it
+// to a specific schema, then return result
+// metaObj - the meta object to be decoded
+// v - the schema defintion to decode to
+// path - the tree path (that is found in the source data document (ERN))
+func DecodeResolverObj(i *Resolver, metaObj *meta.Object, v interface{}, path ...string) (err error) {
+	graph := meta.NewGraph(i.store, metaObj)
+
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("Error decoding %s into %T: %s", path, v, err)
+		}
+	}()
+
+	x, err := graph.Get(path...)
+	if meta.IsPathNotFound(err) {
+		return err
+	} else if err != nil {
+		return err
+	}
+	id, ok := x.(*cid.Cid)
+	if !ok {
+		return fmt.Errorf("Expected %s to be *cid.Cid, got %T", path, x)
+	}
+
+	obj, err := i.store.Get(id)
+	if err != nil {
+		return err
+	}
+	return obj.Decode(v)
+}
+
+/**
+ *	PartyDetails
+*/
+
+// partyDetailsResolver defines grapQL resolver functions for the PartyDetails fields
+type partyDetailsResolver struct {
+	cid          string
+	partyDetails *PartyDetails
+}
+
+func (pd *partyDetailsResolver) Cid() string {
+	return pd.cid
+}
+
+func (pd *partyDetailsResolver) Fullname() string {
+	return pd.partyDetails.PartyName
+}
+
+func (pd *partyDetailsResolver) PartyId() string {
+	return pd.partyDetails.PartyId
+}
+
+// PartyDetails is the resolver function to retrieve 
+// the PartyDetails information from the SQLite index
+func (g *Resolver) PartyDetails(args partyDetailsArgs) ([]*partyDetailsResolver, error) {
+
+	var rows *sql.Rows
+	var err error
+
+	switch {
+	case args.Name != nil:
+		rows, err = g.db.Query("SELECT cid FROM party WHERE name = ?", *args.Name)
+	case args.ID != nil:
+		rows, err = g.db.Query("SELECT cid FROM party WHERE id = ?", *args.ID)
+	default:
+		return nil, errors.New("Missing Name or ID argument in query")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+	var resolvers []*partyDetailsResolver
+	for rows.Next() {
+		var objectID string
+		if err := rows.Scan(&objectID); err != nil {
+			return nil, err
+		}
+		id, err := cid.Parse(objectID)
+		if err != nil {
+			return nil, err
+		}
+
+		obj, err := g.store.Get(id)
+		if err != nil {
+			return nil, err
+		}
+
+		var DdexPartyId struct {
+			Value string `json:"@value"`
+		}
+		if err := DecodeResolverObj(g, obj, &DdexPartyId, "PartyId"); err != nil {
+			pid := &DdexPartyId
+			pid.Value = "000"
```

Comment:
Why are we declaring a variable here? Should we not be returning the error?

Reply:
I dont think it should return an error, hence me returning a dummy ID, however i also agree that it should not be creating this dummy ID. It should return an empty string as the field is not mandatory in the DDEX specification, therefore it should not error, IMO. Happy to discuss. For now i have changed to empty string.